Add SELinux policy module + corresponding specfile changes

fuhry opened the pull request:
Tested in enforcing mode under Fedora 37.

fuhry committed:
add selinux policy module and support for it in the specfile

Conan-Kudo reviewed (changes requested):
The general recommendation is to put SELinux policy modules into a `-selinux` subpackage so you can order things so the SELinux policy is installed _before_ the actual package (which simplifies the installation sequence too).

Conan-Kudo commented:
I've submitted a pull request against this branch with my proposed fixes: https://github.com/fuhry/openssh-ldap-authkeys/pull/8

Conan-Kudo committed:
Restructure RPM packaging spec and rework SELinux module packaging

The SELinux policy module packaging is now reworked to more closely
match the packaging guidelines for packaging independent SELinux
policy modules.

In order to make things more readable, the preamble, scriptlets,
and file lists are now grouped together per binary package.

Conan-Kudo commented on selinux/olak.te:
```diff
@@ -0,0 +1,78 @@
+#
+# other things that need to work on OLAK's stuff
+#
+olak_config_rw(unconfined_t)
+olak_config_rw(puppetagent_exec_t)
+olak_config_rw(puppetagent_t)
```
TIL about this, but is this actually needed? It seems like the Puppet agent policy module says it's supposed to access all domains of config files:

https://github.com/fedora-selinux/selinux-policy/blob/7e50553feb19abeab49911db46c15b50b6bda47e/policy/modules/contrib/puppet.te#L121-L127

fuhry replied:
It seems there's no true way to accomplish inheritance of types; I've aliased `olak_config_t` to `etc_t`.

Conan-Kudo commented on selinux/olak.fc:
```diff
@@ -0,0 +1,8 @@
+# vim: set ts=8 sw=8 noet:
+/usr/bin/openssh-ldap-authkeys		--	gen_context(system_u:object_r:olak_exec_t,s0)
+/etc/openssh-ldap-authkeys		-d	gen_context(system_u:object_r:olak_config_t,s0)
+/etc/openssh-ldap-authkeys/authmap	--	gen_context(system_u:object_r:olak_config_t,s0)
+/etc/openssh-ldap-authkeys/olak.yml	--	gen_context(system_u:object_r:olak_config_t,s0)
```
Couldn't you simplify this to a single rule with `/etc/openssh-ldap-authkeys/.*`?

fuhry replied:
Done - I initially considered setting this context on only the `authmap` and `olak.yml` files but I think it's harmless to set it on everything in this directory.

fuhry committed:
fix post script; pre-generate policy manpage; review fixes

- make olak_config_t inherit from etc_t
- remove now-unused config access macros
- file context definition under /etc/openssh-ldap-authkeys now covers all files in that dir

fuhry committed:
Merge remote-tracking branch 'ngompa/selinux-subpackage' into selinux-module

fuhry committed:
Update Puppet module to account for SELinux package

fuhry committed:
don't generate manpage at build time - it would require the policy to be loaded

fuhry commented:
@Conan-Kudo I've merged your changes and updated the Puppet module to account for them. Would you mind re-reviewing? Thanks!

fuhry committed:
puppet: order selinux module before main package

Conan-Kudo commented on openssh-ldap-authkeys.spec:
```diff
@@ -1,9 +1,13 @@
 Name:		openssh-ldap-authkeys
 Version:	0.2.0
-Release:	0%{?dist}
+Release:	2%{?dist}
```
Was this my fault? Oops?

fuhry replied:
I incremented the release number because this PR in its entirety represents a packaging/distro-specific change.
